module-telemetry/src/main/java/com/mapbox/maps/module/telemetry/MapTelemetryImpl.kt

```diff
@@ -28,39 +34,78 @@ class MapTelemetryImpl : MapTelemetry {
-    val telemetryState: TelemetryEnabler.State =
-      TelemetryEnabler.retrieveTelemetryStateFromPreferences()
-    if (TelemetryEnabler.State.ENABLED == telemetryState) {
-      telemetry.enable()
+
+    // check in the resources
+    val eventsTokenResId = getEventsResId(appContext, EVENTS_ACCESS_TOKEN_RESOURCE_NAME)
+    val eventsToken =
+      if (eventsTokenResId != 0) appContext.getString(eventsTokenResId) else accessToken
+
+    // TODO deferredDeliveryServiceOptions?
+    val eventsServiceOptions = EventsServerOptions(eventsToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
+    this.eventsService = EventsService.getOrCreate(eventsServiceOptions)
+
+    this.telemetryService = TelemetryService.getOrCreate(eventsServiceOptions)
+
+    val coreTelemetryState = TelemetryUtils.getEventsCollectionState()
+    if (coreTelemetryState) {
+      enableTelemetryCollection(true)
     }
   }
 
   /**
-   * Creates a map telemetry instance using telemetry, appplication context and access token
+   * Creates a map telemetry instance using telemetry, application context and access token
    *
-   * @param telemetry the mapbox telemetry
    * @param appContext the application context
    * @param accessToken the mapbox access token
+   * @param eventsService the mapbox EventsServiceInterface
+   * @param telemetryService the mapbox TelemetryService
    */
   @VisibleForTesting
-  constructor(telemetry: MapboxTelemetry, appContext: Context, accessToken: String) {
+  constructor(appContext: Context, accessToken: String, eventsService: EventsServiceInterface, telemetryService: TelemetryService) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = telemetry
+    this.eventsService = eventsService
+    this.telemetryService = telemetryService
   }
 
   /**
    * Register the app user turnstile event
    */
   override fun onAppUserTurnstileEvent() {
-    val turnstileEvent = AppUserTurnstile(
-      BuildConfig.MAPBOX_SDK_IDENTIFIER,
-      BuildConfig.MAPBOX_SDK_VERSION
+    eventsService.sendTurnstileEvent(
+      TurnstileEvent(
+        UserSKUIdentifier.MAPS_MAUS,
+        BuildConfig.MAPBOX_SDK_IDENTIFIER,
+        BuildConfig.MAPBOX_SDK_VERSION
+      ),
+      null
     )
-    turnstileEvent.setSkuId("00") // id MAUS MAPS
-    telemetry.push(turnstileEvent)
-    telemetry.push(MapEventFactory.buildMapLoadEvent(PhoneState(appContext)))
+
+    val mapLoadEvent = MapEventFactory.buildMapLoadEvent(PhoneState(appContext))
+    sendEvent(Gson().toJson(mapLoadEvent))
+  }
+
+  private fun getEventsResId(context: Context, resourceName: String): Int = context.resources.getIdentifier(
+    resourceName,
+    "string",
+    context.packageName
+  )
+
+  private fun sendEvent(event: String) {
+    val eventAttributes = Value.fromJson(event)
+    // TODO: deferredOptions?
+    val mapEvent = eventAttributes.value?.let { Event(EventPriority.IMMEDIATE, it, null) }
```

Comment:
same here, can we just use null for deferredOptions or need some extra logic here? cc @tatiana-yan @tarigo

Reply:
we need `null` here